crates/cow-amm/src/cache.rs

```diff
@@ -85,11 +85,19 @@ impl EventStoring<CowAmmEvent> for Storage {
                 .ok_or_else(|| anyhow::anyhow!("Event missing meta"))?;
             let CowAmmEvent::CowammpoolCreated(cow_amm) = event.data;
             let cow_amm = cow_amm.amm;
+            let amm = match Amm::new(cow_amm, &self.0.helper).await {
+                Ok(amm) => Arc::new(amm),
+                Err(err) if matches!(&err.inner, ExecutionError::Web3(_)) => {
+                    tracing::debug!(?cow_amm, ?err, "retryable error");
+                    return Err(err.into());
+                }
+                Err(err) => {
+                    tracing::info!(?cow_amm, ?err, "helper contract does not support amm");
+                    continue;
+                }
+            };
 
-            cache
-                .entry(meta.block_number)
-                .or_default()
-                .push(Arc::new(Amm::new(cow_amm, &self.0.helper).await?));
+            cache.entry(meta.block_number).or_default().push(amm);
```

Comment:
Could that be a case where some events are processed twice because of a retriable error in the middle, meaning the cache will contain duplicates?

Reply:
Good catch! We would have to be very unlucky to run into this issue but fixing the problem should not be hard either.